### PR TITLE
Import data from firebase to Product Screen

### PR DESCRIPTION
**Pull request by sazzamac**

### CODE REVIEW

**Review by Sajni94 — APPROVED** [review]

File: App.js [hunk 1]
```diff
@@ -19,24 +29,43 @@ const MyTheme = {
...
-            borderBottomWidth: 0,
-          }
-          }}>
-        <Stack.Screen name="ecode" component={Navigation} />
-        <Stack.Screen name="Item" component={ItemScreen}/>
-      </Stack.Navigator>
-    </NavigationContainer>
-  )
+class App extends Component {
+UNSAFE_componentWillMount() {
```
Comment (KejPat) [inline App.js:33]: Is this good to used? Can you use componentDidMount() instead?

File: App.js [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
 import { NavigationContainer, DefaultTheme } from '@react-navigation/native';
 import { createStackNavigator } from '@react-navigation/stack';
-
+import firebase from 'firebase';
 import Navigation from './src/components/Navigation';
+import { LogBox } from 'react-native';
+import _ from 'lodash';
+
+LogBox.ignoreLogs(['Setting a timer']);
+const _console = _.clone(console);
```
Comment (KejPat) [inline App.js:14]: What does this do?

File: src/screens/ProductsScreen.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      getData();
+  }, []);
+
+  // Returned data is an object containing objects - convert to an array of objects
+  const dataArray = _.map(data, (val, uid) => {
+    return {...val, uid};
+});
+
+  // Flatten the nested objects also into arrays and create an array of only the data we want to display
+  const itemsArray = [];
```
Comment (KejPat) [inline src/screens/ProductsScreen.js:29]: this is picky but probably should put initialisations at the top of the function

**Comment by KejPat** [comment]
This is amazing! I don't know how you figured that all out!!

**Comment by KejPat** [comment]
This is an idea, we don't display the number of items, just the item themselves and if they can recycle. If they press recycle a dialog box shows where they can enter the number of items recycle? We can probably limit the number by using the number of items stored in the db and if it is 0 then the recycle will be greyed out

File: App.js (hunk 2 above)
Comment (sazzamac) [reply App.js:14]: It suppresses a warning message that looks really complicated to solve

**Review by KejPat — APPROVED** [review]

File: App.js (hunk 1 above)
Comment (sazzamac) [reply App.js:33]: yep thanks!!

**Pull request merged**